tsl/src/compression/create.c

```diff
@@ -802,9 +802,18 @@ validate_existing_constraints(Hypertable *ht, CompressColInfo *colinfo)
 	{
 		Form_pg_constraint form = (Form_pg_constraint) GETSTRUCT(tuple);
 
-		/* we check primary ,unique and exclusion constraints.
-		 * move foreign key constarints over to compression table
-		 * ignore triggers
+		/* Save away the supporting index, if any, that we check so that we
+		 * can ignore it in the index checking and not get duplicate messages.
+		 *
+		 * We are saving all checked indexes here even though only the unique
+		 * index is a problem at this point. It potentially avoids a second
+		 * check of an index that we have already checked. */
+		if (form->conindid)
```

Comment:
```suggestion
		if (OidIsValid(form->conindid))
```
I guess we should check for a valid oid here, no?

Reply:
Either one works, PostgreSQL use both forms internally. Applied it.